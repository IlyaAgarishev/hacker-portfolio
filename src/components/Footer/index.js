import React, { useRef } from "react";
import styles from "./index.module.css";
import AudioPlayer from "react-h5-audio-player";
import songSrc from "../../audio/song.mp3";
import PropTypes from "prop-types";
import data from "../../data.json";

const Footer = props => {
  const { setPreHacked } = { ...props };
  const song = useRef(null);

  return (
    <div className={styles.footer}>
      <div className={styles.contacts}>
        {data.contacts.map((element, index) => {
          return (
            <a
              href={element.link}
              target="_blank"
              className={styles.link}
              rel="noopener noreferrer"
              style={{ background: element.background, color: "white" }}
              key={index}
            >
              {element.name}
            </a>
          );
        })}
      </div>

      <div className={styles.features}>
        <div
          className={styles.link}
          onClick={() => {
            song.current.togglePlay();
          }}
        >
          song
        </div>

        <div className={styles.link} onClick={() => setPreHacked(true)}>
          skip
        </div>
      </div>
      <AudioPlayer src={songSrc} hidePlayer={true} loop={true} ref={song} />
    </div>
  );
};

Footer.propTypes = {
  setPreHacked: PropTypes.func.isRequired
};

export default Footer;
