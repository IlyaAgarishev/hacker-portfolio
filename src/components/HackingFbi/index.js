import React, { Component } from "react";
import styles from "./index.module.css";
import PropTypes from "prop-types";

class HackingFbi extends Component {
  constructor(props) {
    super(props);
    this.state = { hackingProcess: "", hacked: false };
  }

  componentDidMount() {
    let hackingLoading = setInterval(() => {
      this.setState({
        hackingProcess: this.state.hackingProcess + "☠/"
      });
    }, 200);

    setTimeout(() => {
      clearInterval(hackingLoading);
      this.setState({ hackingProcess: "ACCESS GRANTED" });
      this.setState({ hacked: true });
      this.props.hack();
    }, 2100);
  }

  render() {
    return (
      <div
        onMouseDown={e => {
          this.props.simpleDragAndDrop(this.hackingFbi, e);
        }}
        className={
          this.state.hacked
            ? [styles.hackingFbi, styles.accessGranted].join(" ")
            : styles.hackingFbi
        }
        ref={ref => (this.hackingFbi = ref)}
      >
        <div className={styles.hackingFbiText}>
          {this.state.hacked ? "" : "HACKING FBI DATABASE"}
        </div>
        <div className={styles.hackingProcess}>
          ‍{this.state.hackingProcess}
        </div>
      </div>
    );
  }
}

HackingFbi.propTypes = {
  simpleDragAndDrop: PropTypes.func.isRequired,
  hack: PropTypes.func.isRequired
};

export default HackingFbi;
