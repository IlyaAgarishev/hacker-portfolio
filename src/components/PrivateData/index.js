import React, { Component } from "react";
import styles from "./index.module.css";
import PropTypes from "prop-types";

const PrivateData = props => {
  let privateData;
  const { hacked, simpleDragAndDrop, dndPermission } = { ...props };
  if (hacked) {
    return (
      <div
        className={styles.privateData}
        ref={ref => {
          privateData = ref;
        }}
      >
        <div
          className={styles.privateDataBorder}
          onMouseDown={e => {
            simpleDragAndDrop(privateData, e, dndPermission);
          }}
        />
        <div className={styles.privateDataList}>
          <a
            href="https://github.com/IlyaAgarishev"
            target="_blank"
            rel="noopener noreferrer"
          >
            <img
              className={styles.icon}
              src="https://svgshare.com/i/Aeu.svg"
              alt=""
            />
          </a>
          <a
            href="https://t.me/jamezdean"
            target="_blank"
            rel="noopener noreferrer"
          >
            <img
              className={styles.icon}
              src="https://svgshare.com/i/AdA.svg"
              alt=""
            />
          </a>
          <a
            href="https://api.whatsapp.com/send?phone=89826161073"
            target="_blank"
            rel="noopener noreferrer"
          >
            <img
              className={styles.icon}
              src="https://svgshare.com/i/AeM.svg"
              alt=""
            />
          </a>

          <a
            href="https://www.facebook.com/ilya.agarishev"
            target="_blank"
            rel="noopener noreferrer"
          >
            <img
              className={styles.icon}
              src="https://svgshare.com/i/AeB.svg"
              alt=""
            />
          </a>

          <a
            href="https://vk.com/jamezdean"
            target="_blank"
            rel="noopener noreferrer"
          >
            <img
              className={styles.icon}
              src="https://svgshare.com/i/Acc.svg"
              alt=""
            />
          </a>
        </div>
      </div>
    );
  } else {
    return null;
  }
};

PrivateData.propTypes = {
  simpleDragAndDrop: PropTypes.func.isRequired,
  dndPermission: PropTypes.bool.isRequired,
  hacked: PropTypes.bool.isRequired
};

export default PrivateData;
