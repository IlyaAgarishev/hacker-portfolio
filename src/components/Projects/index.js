import React from "react";
import styles from "./index.module.css";
import PropTypes from "prop-types";

const Projects = props => {
  const { simpleDragAndDrop, dndPermission } = { ...props };
  let projectsWrapper;
  return (
    <div
      className={styles.projectsWrapper}
      ref={ref => (projectsWrapper = ref)}
    >
      <div
        className={styles.projectsBorderTop}
        onMouseDown={e => {
          simpleDragAndDrop(projectsWrapper, e, dndPermission);
        }}
      />
      <div className={styles.projects}>
        <div
          className={styles.projectsText}
          onMouseDown={e => {
            simpleDragAndDrop(projectsWrapper, e, dndPermission);
          }}
        >
          ILYA'S PROJECTS
        </div>
        <div className={styles.projectsList}>
          <a
            href="https://www.sinnlist.com/"
            target="_blank"
            rel="noopener noreferrer"
            className={styles.project}
          >
            sinnlist
          </a>
          <a
            href="https://github.com/IlyaAgarishev/qiqi"
            target="_blank"
            rel="noopener noreferrer"
            className={styles.project}
          >
            qiqi
          </a>
          <a
            href="https://github.com/IlyaAgarishev/naumen-test-task"
            target="_blank"
            rel="noopener noreferrer"
            className={styles.project}
          >
            wiki-search
          </a>
          <a
            href="https://github.com/IlyaAgarishev/react-random-quiz"
            target="_blank"
            rel="noopener noreferrer"
            className={styles.project}
          >
            react-random-quiz
          </a>
          <a
            href="https://github.com/IlyaAgarishev/hacker-portfolio"
            target="_blank"
            rel="noopener noreferrer"
            className={styles.project}
          >
            hacker-portfolio
          </a>
        </div>
      </div>
    </div>
  );
};

Projects.propTypes = {
  simpleDragAndDrop: PropTypes.func.isRequired,
  dndPermission: PropTypes.string.isRequired
};

export default Projects;
