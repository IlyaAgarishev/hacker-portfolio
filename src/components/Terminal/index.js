import React, { useState, useEffect, useRef } from "react";
import styles from "./index.module.css";
import hackscript from "../../hackscript";
import PropTypes from "prop-types";
import { simpleDragAndDrop } from "../../utils";

const Terminal = props => {
  const [textareaValue, setTextareaValue] = useState("");
  const [int, setInt] = useState(0);
  const [times, setTimes] = useState(0);
  const terminal = useRef(null);
  const { setPreHacked, auto } = { ...props };

  const scrollDown = () => {
    terminal.current.scrollTop = terminal.current.scrollHeight;
  };

  useEffect(() => {
    if (!auto) {
      document.onkeydown = () => {
        setTextareaValue(textareaValue + hackscript.slice(int, int + 5));
        setInt(int + 5);
        if (int > hackscript.length) {
          setInt(0);
        } else if (int === 100) {
          setPreHacked(true);
        }
        scrollDown();
      };
    } else {
      if (times !== 3) {
        setTextareaValue(textareaValue + hackscript.slice(int, int + 0.3));
        setInt(int + 0.3);
        if (int > hackscript.length) {
          setInt(0);
          setTimes(times + 1);
        }
        scrollDown();
      }
    }
  });

  return (
    <textarea
      className={
        auto
          ? [styles.terminal, styles.terminalAuto].join(" ")
          : styles.terminal
      }
      placeholder="Start typing..."
      readOnly
      ref={terminal}
      onMouseDown={e => {
        e.preventDefault();
        simpleDragAndDrop(terminal.current, e);
        scrollDown();
      }}
      value={textareaValue}
    />
  );
};

Terminal.propTypes = {
  auto: PropTypes.bool.isRequired
};

export default Terminal;
