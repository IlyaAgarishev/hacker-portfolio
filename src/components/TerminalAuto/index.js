import React, { useState, useEffect, useRef } from "react";
import styles from "./index.module.css";
import codeTwo from "../../codesnippets/code_two.js";
import PropTypes from "prop-types";

const TerminalAuto = props => {
  const [textareaValue, setTextareaValue] = useState("");
  const [int, setInt] = useState(0);
  const [times, setTimes] = useState(0);
  const terminalAuto = useRef(null);
  const { simpleDragAndDrop } = { ...props };

  const speed = 0.5;

  useEffect(() => {
    if (times !== 3) {
      setTextareaValue(textareaValue + codeTwo.slice(int, int + speed));
      setInt(int + speed);
      if (int > codeTwo.length) {
        setInt(0);
        setTimes(times + 1);
        console.log(times);
      }
      terminalAuto.current.scrollTop = terminalAuto.current.scrollHeight;
    }
  });

  return (
    <textarea
      className={styles.terminalAuto}
      placeholder={"Start typing..."}
      ref={terminalAuto}
      readOnly
      onMouseDown={e => {
        e.preventDefault();
        simpleDragAndDrop(terminalAuto.current, e);
      }}
      value={textareaValue}
    />
  );
};

TerminalAuto.propTypes = {
  simpleDragAndDrop: PropTypes.func.isRequired
};

export default TerminalAuto;
