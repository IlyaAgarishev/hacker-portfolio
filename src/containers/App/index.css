.footer {
  position: fixed;
  bottom: 0;
  width: 100%;
  height: 30px;
  background: rgb(0, 255, 0);
  display: flex;
  align-items: center;
}

.github-link {
  height: 70%;
  border-radius: 2px;
  margin-left: 2px;
  display: flex;
  cursor: pointer;
  justify-content: center;
  align-items: center;
  width: 70px;
  text-decoration: none;
  border: 1px solid black;
  color: black;
  transition: 0.2s all ease;
}

.github-link:hover {
  background: rgb(0, 241, 0);
}

.globe {
  width: 150px;
  position: absolute;
  top: 45%;
  left: 45%;
}

.ilyahacker {
  width: 400px;
  position: absolute;
  left: 45%;
  margin-top: 10px;
  border: 1px solid rgb(255, 255, 255);
  border-top: 20px solid rgb(255, 255, 255);
}
