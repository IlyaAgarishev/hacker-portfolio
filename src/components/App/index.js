import React, { useState, useEffect, useRef } from "react";
import Terminal from "../Terminal";
import HackingFbi from "../HackingFbi";
import Projects from "../Projects";
import ilyaSvg from "../../img/ilyahacker.svg";
import styles from "./index.module.css";
import Footer from "../Footer";
import { simpleDragAndDrop } from "../../utils";

const App = () => {
  const [preHacked, setPreHacked] = useState(false);
  const [hacked, setHacked] = useState(false);
  const [dndPermission, setDndPermission] = useState("granted");
  const ilyahacker = useRef(null);

  useEffect(() => {
    if (window.innerWidth < 975) {
      setHacked(true);
      setDndPermission("denied");
    }
  });

  return (
    <div className={styles.app}>
      <img
        src="https://i.gifer.com/C6Zz.gif"
        alt="globe"
        className={styles.globe}
      />

      <Terminal setPreHacked={setPreHacked} auto={false} />

      {preHacked === true && (
        <div>
          <Terminal auto={true} />
          <HackingFbi setHacked={setHacked} hacked={hacked} />
        </div>
      )}

      {hacked === true && (
        <div>
          <img
            src={ilyaSvg}
            alt=""
            className={styles.ilyahacker}
            ref={ilyahacker}
            onMouseDown={e => {
              simpleDragAndDrop(ilyahacker.current, e, dndPermission);
            }}
          />

          <Projects dndPermission={dndPermission} />
        </div>
      )}

      <Footer setPreHacked={setPreHacked} />
    </div>
  );
};

export default App;
