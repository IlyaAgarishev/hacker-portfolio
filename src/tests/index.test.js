import React from "react";
import { mount, render, shallow } from "enzyme";
import App from "../components/App";
import Terminal from "../components/Terminal";
import PrivateData from "../components/PrivateData";
import Footer from "../components/Footer";
import { simpleDragAndDrop, getCoords } from "../dnd.js";
import TerminalAuto from "../components/TerminalAuto";
import HackingFbi from "../components/HackingFbi";
import Projects from "../components/Projects";

// Functions testing

test("getCoords returns object", () => {
  const div = document.createElement("div");
  expect(typeof getCoords(div) === "object").toBeTruthy();
});

test("getCoords returns object with properties left and top", () => {
  const div = document.createElement("div");
  expect(getCoords(div).left).not.toBeUndefined();
  expect(getCoords(div).top).not.toBeUndefined();
});

test("simple dnd doesn't return anything by default", () => {
  const div = document.createElement("div");
  const e = { pageX: 0, pageY: 0 };
  expect(simpleDragAndDrop(div, e)).toBeUndefined();
});

test("simple dnd reurns null if dndPermission is denied", () => {
  const div = document.createElement("div");
  const e = { pageX: 0, pageY: 0 };
  expect(simpleDragAndDrop(div, e, "denied")).toBeNull();
});

// Components testing

test("App", () => {
  const component = shallow(<App />);
  expect(component).toMatchSnapshot();
});

test("Footer", () => {
  const component = shallow(<Footer />);
  expect(component).toMatchSnapshot();
});

test("Terminal", () => {
  const component = shallow(
    <Terminal simpleDragAndDrop={jest.fn()} setHackInterval={jest.fn()} />
  );
  expect(component).toMatchSnapshot();
});

test("Terminal Auto", () => {
  const component = shallow(<TerminalAuto simpleDragAndDrop={jest.fn()} />);
  expect(component).toMatchSnapshot();
});

test("Hacking Fbi", () => {
  const component = shallow(
    <HackingFbi simpleDragAndDrop={jest.fn()} hack={jest.fn()} />
  );
  expect(component).toMatchSnapshot();
});

test("Projects", () => {
  const component = shallow(
    <Projects
      simpleDragAndDrop={jest.fn()}
      dndPermission={jest.fn()}
      hacked={jest.fn()}
    />
  );
  expect(component).toMatchSnapshot();
});

test("Private Data", () => {
  const component = shallow(
    <PrivateData
      simpleDragAndDrop={jest.fn()}
      dndPermission={jest.fn()}
      hacked={jest.fn()}
    />
  );
  expect(component).toMatchSnapshot();
});
