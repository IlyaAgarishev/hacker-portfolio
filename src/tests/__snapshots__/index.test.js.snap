// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App 1`] = `
<div
  className="app"
>
  <img
    alt="globe"
    className="globe"
    src="https://i.gifer.com/C6Zz.gif"
  />
  <Terminal
    setHackInterval={[Function]}
    simpleDragAndDrop={[Function]}
  />
  <Projects
    dndPermission="granted"
    hacked={false}
    simpleDragAndDrop={[Function]}
  />
  <PrivateData
    dndPermission="granted"
    hacked={false}
    simpleDragAndDrop={[Function]}
  />
  <Footer />
</div>
`;

exports[`Footer 1`] = `
<div
  className="footer"
>
  <a
    className="githubLink"
    href="https://github.com/IlyaAgarishev/hacker-portfolio"
    rel="noopener noreferrer"
    target="_blank"
  >
    github
  </a>
  <div
    className="playSong"
    onClick={[Function]}
  >
    song
  </div>
  <H5AudioPlayer
    autoPlay={false}
    hidePlayer={true}
    listenInterval={1000}
    loop={true}
    muted={false}
    preload="auto"
    progressUpdateInterval={500}
    src="song.mp3"
    volume={1}
  />
</div>
`;

exports[`Hacking Fbi 1`] = `
<div
  className="hackingFbi"
  onMouseDown={[Function]}
>
  <div
    className="hackingFbiText"
  >
    HACKING FBI DATABASE
  </div>
  <div
    className="hackingProcess"
  >
    ‚Äç
  </div>
</div>
`;

exports[`Private Data 1`] = `
<div
  className="privateData"
>
  <div
    className="privateDataBorder"
    onMouseDown={[Function]}
  />
  <div
    className="privateDataList"
  >
    <a
      href="https://github.com/IlyaAgarishev"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="github"
        className="icon"
        src="https://svgshare.com/i/Aeu.svg"
      />
    </a>
    <a
      href="https://t.me/jamezdean"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="telegram"
        className="icon"
        src="https://svgshare.com/i/AdA.svg"
      />
    </a>
    <a
      href="https://api.whatsapp.com/send?phone=89826161073"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="whatsapp"
        className="icon"
        src="https://svgshare.com/i/AeM.svg"
      />
    </a>
    <a
      href="https://www.facebook.com/ilya.agarishev"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="facebook"
        className="icon"
        src="https://svgshare.com/i/AeB.svg"
      />
    </a>
    <a
      href="https://vk.com/jamezdean"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="vk"
        className="icon"
        src="https://svgshare.com/i/Acc.svg"
      />
    </a>
  </div>
</div>
`;

exports[`Projects 1`] = `
<div
  className="projectsWrapper"
>
  <div
    className="projectsBorderTop"
    onMouseDown={[Function]}
  />
  <div
    className="projects"
  >
    <div
      className="projectsText"
      onMouseDown={[Function]}
    >
      ILYA'S PROJECTS
    </div>
    <div
      className="projectsList"
    >
      <a
        className="project"
        href="https://www.sinnlist.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        sinnlist
      </a>
      <a
        className="project"
        href="https://github.com/IlyaAgarishev/qiqi"
        rel="noopener noreferrer"
        target="_blank"
      >
        qiqi
      </a>
      <a
        className="project"
        href="https://github.com/IlyaAgarishev/naumen-test-task"
        rel="noopener noreferrer"
        target="_blank"
      >
        wiki-search
      </a>
      <a
        className="project"
        href="https://github.com/IlyaAgarishev/react-random-quiz"
        rel="noopener noreferrer"
        target="_blank"
      >
        react-random-quiz
      </a>
      <a
        className="project"
        href="https://github.com/IlyaAgarishev/hacker-portfolio"
        rel="noopener noreferrer"
        target="_blank"
      >
        hacker-portfolio
      </a>
    </div>
  </div>
</div>
`;

exports[`Terminal 1`] = `
<textarea
  className="terminal"
  onMouseDown={[Function]}
  placeholder="Start typing..."
  readOnly={true}
  value=""
/>
`;

exports[`Terminal Auto 1`] = `
<textarea
  className="terminalAuto"
  onMouseDown={[Function]}
  placeholder="Start typing..."
  readOnly={true}
  value=""
/>
`;
